api/src/main/java/com/capitalone/dashboard/config/LoggingFilter.java

```diff
@@ -0,0 +1,464 @@
+                .append(bufferedRequest.getRequestBody())
+                .append("] [REMOTE:")
+                .append(httpServletRequest.getRemoteAddr())
+                .append("] [STATUS:")
+                .append(bufferedResponse.getStatus())
+                .append("]");
+        LOGGER.info(logMessage.toString());
+
+        if (httpServletRequest.getMethod().equals(HttpMethod.PUT.toString()) ||
+                (httpServletRequest.getMethod().equals(HttpMethod.POST.toString())) ||
+                (httpServletRequest.getMethod().equals(HttpMethod.DELETE.toString()))) {
+            requestLogRepository.save(requestLog);
+        }
+    }
+
+
+    private Map<String, String> getTypesafeRequestMap(HttpServletRequest request) {
+        Map<String, String> typesafeRequestMap = new HashMap<String, String>();
+        Enumeration<?> requestParamNames = request.getParameterNames();
+        if (requestParamNames != null) {
+            while (requestParamNames.hasMoreElements()) {
+                String requestParamName = (String) requestParamNames.nextElement();
+                String requestParamValue = request.getParameter(requestParamName);
+                typesafeRequestMap.put(requestParamName, requestParamValue);
+            }
+        }
+        return typesafeRequestMap;
+    }
+
+
+    @Override
+    public void destroy() {
+    }
+
+
+    private static final class BufferedRequestWrapper extends HttpServletRequestWrapper {
+
+        private ByteArrayInputStream bais = null;
+        private ByteArrayOutputStream baos = null;
+        private BufferedServletInputStream bsis = null;
+        private byte[] buffer = null;
+
+
+        public BufferedRequestWrapper(HttpServletRequest req) throws IOException {
+            super(req);
+            // Read InputStream and store its content in a buffer.
+            InputStream is = req.getInputStream();
+            this.baos = new ByteArrayOutputStream();
+            byte buf[] = new byte[1024];
+            int letti;
+            while ((letti = is.read(buf)) > 0) {
+                this.baos.write(buf, 0, letti);
+            }
+            this.buffer = this.baos.toByteArray();
+        }
+
+
+        @Override
+        public ServletInputStream getInputStream() {
+            this.bais = new ByteArrayInputStream(this.buffer);
+            this.bsis = new BufferedServletInputStream(this.bais);
+            return this.bsis;
+        }
+
+
+        String getRequestBody() throws IOException {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(this.getInputStream()));
+            String line = null;
+            StringBuilder inputBuffer = new StringBuilder();
+            do {
+                line = reader.readLine();
+                if (null != line) {
+                    inputBuffer.append(line.trim());
+                }
+            } while (line != null);
+            reader.close();
+            return inputBuffer.toString().trim();
```

Comment:
at this point there are several copies of the body in memory.. we should rethink & optimize this..